src/main/kotlin/nl/info/zac/documentcreation/DocumentCreationService.kt

```diff
@@ -5,128 +5,54 @@
 package nl.info.zac.documentcreation
 
 import jakarta.enterprise.context.ApplicationScoped
-import jakarta.enterprise.inject.Instance
 import jakarta.inject.Inject
 import jakarta.ws.rs.core.UriBuilder
 import net.atos.client.zgw.zrc.model.Zaak
 import net.atos.client.zgw.zrc.model.ZaakInformatieobject
 import net.atos.zac.util.MediaTypes
-import nl.info.client.smartdocuments.model.document.OutputFormat
-import nl.info.client.smartdocuments.model.document.Selection
-import nl.info.client.smartdocuments.model.document.SmartDocument
-import nl.info.client.smartdocuments.model.document.Variables
 import nl.info.zac.app.informatieobjecten.EnkelvoudigInformatieObjectUpdateService
-import nl.info.zac.authentication.LoggedInUser
 import nl.info.zac.configuratie.ConfiguratieService
 import nl.info.zac.documentcreation.converter.DocumentCreationDataConverter
-import nl.info.zac.documentcreation.model.DocumentCreationAttendedResponse
-import nl.info.zac.documentcreation.model.DocumentCreationDataAttended
-import nl.info.zac.identity.model.getFullName
 import nl.info.zac.smartdocuments.SmartDocumentsService
-import nl.info.zac.smartdocuments.SmartDocumentsTemplatesService
 import nl.info.zac.util.AllOpen
 import nl.info.zac.util.NoArgConstructor
 import java.net.URI
 import java.time.ZonedDateTime
 import java.time.format.DateTimeFormatter
 import java.util.UUID
-import java.util.logging.Level
-import java.util.logging.Logger
 
 @NoArgConstructor
 @ApplicationScoped
 @AllOpen
 @Suppress("LongParameterList")
 class DocumentCreationService @Inject constructor(
```

Comment:
Shouldn't this be an abstract class? Also not @ApplicationScoped?

Not sure I like this inheritance construction on services. I would not do that to be honest. Better than to add an extra service layer underneath if you want to share functionality between the CMMN and BPMN worlds I think. 

The general consensus today is to prefer composition over inheritance, especially for such service classes. Which is also partly why in Kotlin classes are final by default.

Reply:
https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
Extracted this in a separate service to enable injection of parameters and split tests